modules/c++/mem/include/mem/BufferView.h

```diff
@@ -50,6 +53,34 @@ struct BufferView
     T* data;
     size_t size;
 
+    // returns a new bufferView that "takes" sectionSize T from the bufferView
+    // For instance, if the current bufferView had 100 ints, and you called 
+    // section with sectionSize = 10, section() would return a new BufferView
+    // of size 10 and the original BufferView would be left with 90 ints.
+    // throws std::string if there is not enough space 
```

Comment:
Update comment to reflect what type is thrown

Also, use Doxygen-style comments (I will admit we have often been lazy about this).  So use stuff like ```\throws``` to describe the exception